src/lib.rs

```diff
@@ -1110,6 +1110,64 @@ impl<K, V, S> OrderMap<K, V, S>
         });
     }
 
+    /// Sort the map’s key-value pairs by the default ordering of the keys.
+    pub fn sort_keys(&mut self)
+        where K: Ord,
+    {
+        self.sort_by(|k1, _, k2, _| Ord::cmp(k1, k2))
+    }
+
+    /// Sort the map’s key-value pairs in place using the comparison
+    /// function `compare`.
+    ///
+    /// The comparison function receives two key and value pairs to compare (you
+    /// can sort by keys or values or their combination as needed).
+    pub fn sort_by<F>(&mut self, mut compare: F)
+        where F: FnMut(&K, &V, &K, &V) -> Ordering,
+    {
+        // new_index will form a lookup map from current index -> new index.
+        let mut new_index = Vec::from_iter(0..self.len());
+        new_index.sort_by(|&i, &j| {
+            let ei = &self.entries[i];
+            let ej = &self.entries[j];
+            compare(&ei.key, &ei.value, &ej.key, &ej.value)
```

Comment:
I measured and the gain from using unchecked indexing is up to 10% here for getting `ei`, `ej`. Just FYI.